src/python/pants/init/specs_calculator.py

```diff
@@ -55,10 +55,16 @@ def calculate_specs(
         raise InvalidSpecConstraint(
             "The `--changed-*` options are only available if Git is used for the repository."
         )
-    changed_request = ChangedRequest(
-        sources=tuple(changed_options.changed_files(maybe_git_worktree.git_worktree)),
-        dependees=changed_options.dependees,
-    )
+
+    changed_files = tuple(changed_options.changed_files(maybe_git_worktree.git_worktree))
+
+    # We input all changed files as file arguments, which makes sure that target-less goals like
+    # `count-loc` recognize all files that have changed, not just ones owned by targets.
+    file_literal_specs = tuple(FileLiteralSpec(f) for f in changed_files)
+
+    # We must no matter what also find the target owners to handle deleted targets, along with
+    # `--changed-dependees`.
```

Comment:
It's not clear what this comment is referring to... seems like it should maybe be on the `Specs` constructor instead, to justify including both file and address specs?